tests/robustness/failpoint/network.go

```diff
@@ -57,6 +57,9 @@ func (tb triggerBlackhole) Trigger(ctx context.Context, t *testing.T, member e2e
 }
 
 func (tb triggerBlackhole) Available(config e2e.EtcdProcessClusterConfig, process e2e.EtcdProcess) bool {
+	if tb.waitTillSnapshot && config.ServerConfig.SnapshotCatchUpEntries > 100 {
```

Comment:
Test will not pass if SnapshotCatchUpEntries is not set to lower values, default in older releases was 10`000, which is much too much. We cannot allow failpoint randomiser to pick it.

Reply:
Is the intention here to only manually hardcode a failpoint with a `triggerBlackhole` trigger and never to allow `PickRandom` to pick it? Because I see only this where we configure `SnapshotCatchupEntried`:
https://github.com/etcd-io/etcd/blob/333bd7bf7612a9f83cc94e8334565012e84f354f/tests/robustness/scenarios.go#L175

This sets it to `100`. And I also see `waitTillSnapshot` is always set to `true` in whatever failpoint that is defined with a `triggerBlackhole` trigger.

If this is the intention can you add a comment? Also, is it okay disallowing picking a failpoint with `triggerBlackhole` for all failpoints?
Maybe as a followup, we can disallow a particular scenario from being considered by `PickRandom`.

Reply:
Intention is to revert change from https://github.com/etcd-io/etcd/pull/17967/files where this was already present.

Reply:
Yeah, my bad for not clarifying. I meant to ask this irrespective of the revert.

Reply:
Added a comment

Reply:
Also, the default for `SnapshotCatchupEntries` always seems to be `5000` and for `DefaultSnapshotCount` is `10000`. I'm guessing `5000` is too high as well?

Reply:
Right, however in robustness test we lower snapshot count to 50-1000.